### ISSUE TITLE
Doc improvement request: Base signal

### ISSUE THREAD

**Issue by woozey** [hyperspy/hyperspy#1932]
At the moment in `init` of `BaseSignal` parameter `axes` described as follows:
```python
axes : dictionary (optional)
    Dictionary to define the axes (see the
    documentation of the AxesManager class for more details).
```
However, in `AxesManager` documentation dictionary syntax is either not mentioned or I [user] couldn't [won't be able to] find it.

**Comment by thomasaarholt** [comment]
I would appreciate this too :) I always end up in a loop trying to figure out how one should initialise axes.

**Comment by ericpre** [comment]
This refers to the docstring of the `AxesManager` class, which you can access it with `hyperspy.axes.AxesManager(` or `hyperspy.axes.AxesManager?` in a interactive environment. This is in the [`axes`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/axes.py#L466) module and I would say that the docstring of the `__init__` of `BaseSignal` is rather for developer than for users.

There is also an example in the [user guide](http://hyperspy.readthedocs.io/en/stable/user_guide/tools.html#signal-initialization) which explain how to create a signal with custom axes (even if `dict1` is not defined in this example...). Is this example not enough for what you want to do?

In any case, the docstring of all the signals is quite bad at the moment.

**Comment by thomasaarholt** [comment]
Thank you for clarifying. It would be great if someone could clean things up. I haven't got time at the moment, so perhaps someone else can? Cleaning up just a bit at a time makes things easier.

**Comment by woozey** [comment]
@ericpre 
> This refers to the docstring of the AxesManager class, which you can access it with hyperspy.axes.AxesManager( or hyperspy.axes.AxesManager? in a interactive environment.

Yes, one can access it if one imports hyperspy (not api). But as mentioned above it doesn't describe how the dict should look like. The same doc is accessible via interactive help of `s.axes_manager`.
> and I would say that the docstring of the `__init__` of `BaseSignal` is rather for developer than for users.

This is actually what is displayed in interactive help when I do following:
``` python
hs.signals.Signal1D(data, )
```
This is why I thought it would worth improving it...
> There is also an example in the user guide which explain how to create a signal with custom axes (even if dict1 is not defined in this example...). Is this example not enough for what you want to do?

Yep, the example is indeed the most helpful thing here. And yes it does what I wanted to do, i.e. to  pass `'navigation':True` or `False` in order to say which axes should have which role (see discussion on gitter).
``` python
>>> dict0 = {'size': 2, 'navigate':True}
>>> dict1 = {'size': 3, 'navigate':True}
>>> dict2 = {'size': 4, 'navigate':False}
>>> dict3 = {'size': 5, 'navigate':True}
>>> hs.signals.Signal1D(np.random.rand(120).reshape((2,3,4,5)), axes=[dict0, dict1, dict2, dict3])
<Signal1D, title: , dimensions: (5, 3, 2|4)>
```
Though, odd thing is that `size` entry in the dicts is required... Which in the case of bad use leads to:
``` python
>>> dict0 = {'size': 2, 'navigate':True}
>>> dict1 = {'size': 3, 'navigate':True}
>>> dict2 = {'size': 6, 'navigate':False}
>>> dict3 = {'size': 5, 'navigate':True}
>>> s = hs.signals.Signal1D(np.random.rand(120).reshape((2,3,4,5)), axes=[dict0, dict1, dict2, dict3])
>>> s
<Signal1D, title: , dimensions: (5, 3, 2|6)>
>>> s.inav[0, 0, 0].data.shape
(4,)
```
Maybe this is the moment when user should get a warning...
